File: server/src/main/java/dev/dtos/responses/OrderResponse.java
package dev.dtos.responses;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
@Builder
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class OrderResponse {
    private String id;
    private Double total;
    private String status;
    private String shipping_address;
    private Date createdAt;
    private Date updatedAt;
    private String paymentMethod;
    private String shippingMethod;
    private List<OrderItemsResponse> orderedItems;
}

File: mobile/app/src/main/java/com/javaproject/myapplication/data/model/response/OrderResponse.kt
package com.javaproject.myapplication.data.model.response

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class OrderResponse (
    @SerializedName("id")
    val id: String,

    @SerializedName("total")
    val total: Double,

    @SerializedName("status")
    val status: String,

    @SerializedName("shipping_address")
    val shippingAddress: String,

    @SerializedName("createdAt")
    val createdAt: LocalDateTime,

    @SerializedName("updatedAt")
    val updatedAt: LocalDateTime,

    @SerializedName("paymentMethod")
    val paymentMethod: String,

    @SerializedName("shippingMethod")
    val shippingMethod: String,

    @SerializedName("orderedItems")
    val orderedItems: List<OrderedItemResponse>
)